asm/osSetIntMask.s

```diff
@@ -1,123 +1,131 @@
+    .half MI_INTR_MASK_SET_SP | MI_INTR_MASK_CLR_SI | MI_INTR_MASK_CLR_AI | MI_INTR_MASK_SET_VI | MI_INTR_MASK_SET_PI | MI_INTR_MASK_SET_DP
+    .half MI_INTR_MASK_CLR_SP | MI_INTR_MASK_SET_SI | MI_INTR_MASK_CLR_AI | MI_INTR_MASK_SET_VI | MI_INTR_MASK_SET_PI | MI_INTR_MASK_SET_DP
+    .half MI_INTR_MASK_SET_SP | MI_INTR_MASK_SET_SI | MI_INTR_MASK_CLR_AI | MI_INTR_MASK_SET_VI | MI_INTR_MASK_SET_PI | MI_INTR_MASK_SET_DP
+    .half MI_INTR_MASK_CLR_SP | MI_INTR_MASK_CLR_SI | MI_INTR_MASK_SET_AI | MI_INTR_MASK_SET_VI | MI_INTR_MASK_SET_PI | MI_INTR_MASK_SET_DP
+    .half MI_INTR_MASK_SET_SP | MI_INTR_MASK_CLR_SI | MI_INTR_MASK_SET_AI | MI_INTR_MASK_SET_VI | MI_INTR_MASK_SET_PI | MI_INTR_MASK_SET_DP
+    .half MI_INTR_MASK_CLR_SP | MI_INTR_MASK_SET_SI | MI_INTR_MASK_SET_AI | MI_INTR_MASK_SET_VI | MI_INTR_MASK_SET_PI | MI_INTR_MASK_SET_DP
+    .half MI_INTR_MASK_SET_SP | MI_INTR_MASK_SET_SI | MI_INTR_MASK_SET_AI | MI_INTR_MASK_SET_VI | MI_INTR_MASK_SET_PI | MI_INTR_MASK_SET_DP
+ENDDATA(__osRcpImTable)
 
-.section .rodata
+.section .text
 
 .balign 16
 
-glabel __osRcpImTable
-    .half 0x0555
-    .half 0x0556
-    .half 0x0559
-    .half 0x055A
-    .half 0x0565
-    .half 0x0566
-    .half 0x0569
-    .half 0x056A
-    .half 0x0595
-    .half 0x0596
-    .half 0x0599
-    .half 0x059A
-    .half 0x05A5
-    .half 0x05A6
-    .half 0x05A9
-    .half 0x05AA
-    .half 0x0655
-    .half 0x0656
-    .half 0x0659
-    .half 0x065A
-    .half 0x0665
-    .half 0x0666
-    .half 0x0669
-    .half 0x066A
-    .half 0x0695
-    .half 0x0696
-    .half 0x0699
-    .half 0x069A
-    .half 0x06A5
-    .half 0x06A6
-    .half 0x06A9
-    .half 0x06AA
-    .half 0x0955
-    .half 0x0956
-    .half 0x0959
-    .half 0x095A
-    .half 0x0965
-    .half 0x0966
-    .half 0x0969
-    .half 0x096A
-    .half 0x0995
-    .half 0x0996
-    .half 0x0999
-    .half 0x099A
-    .half 0x09A5
-    .half 0x09A6
-    .half 0x09A9
-    .half 0x09AA
-    .half 0x0A55
-    .half 0x0A56
-    .half 0x0A59
-    .half 0x0A5A
-    .half 0x0A65
-    .half 0x0A66
-    .half 0x0A69
-    .half 0x0A6A
-    .half 0x0A95
-    .half 0x0A96
-    .half 0x0A99
-    .half 0x0A9A
-    .half 0x0AA5
-    .half 0x0AA6
-    .half 0x0AA9
-    .half 0x0AAA
+LEAF(osSetIntMask)
+    mfc0    $t4, C0_SR
+    andi    $v0, $t4, (SR_IMASK | SR_IE)
+    lui     $t0, %hi(__OSGlobalIntMask)
+    addiu   $t0, %lo(__OSGlobalIntMask)
+    lw      $t3, ($t0)
+    li      $at, -1
+    xor     $t0, $t3, $at
+    andi    $t0, $t0, SR_IMASK
+    or      $v0, $v0, $t0
+    lui     $t2, %hi(PHYS_TO_K1(MI_INTR_MASK_REG))
+    lw      $t2, %lo(PHYS_TO_K1(MI_INTR_MASK_REG))($t2)
+    beqz    $t2, 1f
+     srl    $t1, $t3, 0x10
+    li      $at, -1
+    xor     $t1, $t1, $at
+    andi    $t1, $t1, MI_INTR_MASK
+    or      $t2, $t2, $t1
+1:
+    sll     $t2, $t2, 0x10
+    or      $v0, $v0, $t2
+    lui     $at, MI_INTR_MASK
+    and     $t0, $a0, $at
+    and     $t0, $t0, $t3
+    srl     $t0, $t0, 0xf
```

Comment:
small inconsistencies among .s files, not a big deal:

the shift being hexadecimal or decimal in srl/sll instructions
hexadecimal being lower/upper case

Reply:
I've changed all the hex to use uppercase since as I understand it that's our convention for C files also. The discrepancy between hex and dec in shifts is due to <= 9 not being prefixed by 0x.